stdlib/public/Python/PythonLibrary.swift

```diff
@@ -0,0 +1,200 @@
+//===-- PythonLibrary.swift -----------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements the logic for dynamically loading Python at runtime.
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(Darwin)
+import Darwin
+#elseif canImport(Glibc)
+import Glibc
+#endif
+
+//===----------------------------------------------------------------------===//
+// The `PythonLibrary` struct that loads Python symbols at runtime.
+//===----------------------------------------------------------------------===//
+
+struct PythonLibrary {
+  
+  private static let shared = PythonLibrary()
+  private static let pythonLegacySymbolName = "PyString_AsString"
+  
+  private let pythonLibraryHandle: UnsafeMutableRawPointer
+  private let isLegacyPython: Bool
+  
+  fileprivate init() {
+    guard let pythonLibraryHandle = PythonLibrary.getPythonLibraryHandle() else {
+      fatalError(
+        "Python library not found. Set the \(Environment.library.key) " +
+        "environment variable with the path to the Python Library."
+      )
+    }
+    
+    self.pythonLibraryHandle = pythonLibraryHandle
+    
+    // Check if Python is legacy (Python 2)
+    self.isLegacyPython = dlsym(pythonLibraryHandle, PythonLibrary.pythonLegacySymbolName) != nil
+    
+    if self.isLegacyPython {
+      PythonLibrary.log(message: "Loaded legacy Python library, using legacy symbols...")
+    }
+  }
+  
+  static func getSymbol<T>(name: String, legacyName: String? = nil, signature: T.Type) -> T {
+    var name = name
+    
```

Comment:
Please remove empty lines here and everywhere else between trivial statements to make things more compact.